### ISSUE TITLE
providers.file: Changes in bind-mounted dynamic config file are ignored until restart

### ISSUE THREAD

**Issue by sgofferj** [containous/traefik#5589]
### Do you want to request a *feature* or report a *bug*?

Bug (probably)

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?
I configured a dynamic config file `dynamic.toml` with `watch = true`. The config file is a bind mount in a docker container.

### What did you expect to see?
I expect traefik to notice changes in `dynamic.toml` within some seconds and apply them.

### What did you see instead?
traefik ignores any changes. A restart is necessary to apply changes in `dynamic.toml`

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      2.0.1
Codename:     montdor
Go version:   go1.13.1
Built:        2019-09-26T16:18:03Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Xeon E3 1240L v3 / 16GB / Ubuntu Server 16.04 / Docker
 
docker-compose.yml:
```
version: '2'                                                                                                                        
                                                                                                                                    
services:                                                                                                                           
  traefik:                                                                                                                          
    image: traefik:v2.0                                                                                                             
    ports:                                                                                                                          
      - "80:80"                                                                                                                     
      - "443:443"                                                                                                                   
    volumes:                                                                                                                        
      - /var/run/docker.sock:/var/run/docker.sock                                                                                   
      - /storage/docker/traefik/acme.json:/etc/traefik/acme/acme.json                                                               
      - /storage/docker/traefik/traefik.toml:/etc/traefik/traefik.toml                                                              
      - /storage/docker/traefik/dynamic.toml:/etc/traefik/dynamic.toml                                                              
      - /etc/letsencrypt/live/gofferje.net/fullchain.pem:/etc/traefik/gn_fullchain.pem                                              
      - /etc/letsencrypt/live/gofferje.net/privkey.pem:/etc/traefik/gn_privkey.pem                                                  
    networks:                                                                                                                       
      - web_services                                                                                                                
                                                                                                                                    
networks:                                                                                                                           
  web_services:                                                                                                                     
    external:                                                                                                                       
      name: web_services                                                                                                            
```

traefik.toml (excerpt):
```
[providers.docker]
  exposedByDefault = false

[providers.file]
  filename = "/etc/traefik/dynamic.toml"
  watch = true
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

No relevant log output. File changes are simply ignored.

**Comment by MJPA** [comment]
Coming across this too. Even starting a shell in the traefik container and editing the file with vi doesn't get traefik to update.

**Comment by ldez** [comment]
The workaround is to mount and use a directory instead of a file..

**Comment by MJPA** [comment]
That doesn't fully work for me. It only refreshes the config if a new file is created, if an existing file is changed, the config is not reloaded either.

**Comment by ldez** [comment]
```
[providers.docker]
  exposedByDefault = false

[providers.file]
  filename = "/etc/traefik/dynamic/"
  watch = true
```

**Comment by MJPA** [comment]
That's what I have (but `directory` instead of `filename`). If `/etc/traefik/dynamic/` contains `foo.yml` when traefik starts, modifying `foo.yml` does nothing at all to the config, but a simple `touch bar.yml` and it picks up changes to both files.

I've changed my code now so that it removes the dynamically generated file and creates a new one and traefik is picking it up, but just overwriting the file and it doesn't, even with watching a directory.

**Comment by ldez** [comment]
you also need to mound the directory instead a file. ^

**Comment by MJPA** [comment]
I do, my traefik.yml contains:

    providers:
      file:
        directory: "/etc/traefik/dynamic/"
        watch: true

I have a bind mount for `./traefik-dynamic/:/etc/traefik/dynamic/`. That directory contains `generated.yml`. If I edit that file, traefik doesn't pick up the changes. I have to either delete it first and write to it (some editors do this silently) or create a 2nd file in the mounted / watched directory.

**Comment by traefiker** [comment]
Closed by #5939.
